src/com/goide/inspections/GoMultiplePackagesQuickFix.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright 2013-2015 Sergey Ignatov, Alexander Zolotov, Mihai Toader, Florin Patan
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.goide.inspections;
+
+import com.goide.GoConstants;
+import com.goide.psi.GoFile;
+import com.goide.psi.GoPackageClause;
+import com.goide.psi.impl.GoElementFactory;
+import com.goide.runconfig.testing.GoTestFinder;
+import com.goide.util.GoUtil;
+import com.intellij.codeInspection.LocalQuickFixAndIntentionActionOnPsiElement;
+import com.intellij.openapi.command.WriteCommandAction;
+import com.intellij.openapi.editor.Editor;
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.ui.popup.JBPopup;
+import com.intellij.openapi.ui.popup.JBPopupFactory;
+import com.intellij.psi.PsiDirectory;
+import com.intellij.psi.PsiElement;
+import com.intellij.psi.PsiFile;
+import com.intellij.ui.components.JBList;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import javax.swing.*;
+import java.util.Collection;
+
+public class GoMultiplePackagesQuickFix extends LocalQuickFixAndIntentionActionOnPsiElement {
+  private Collection<String> myPackages;
+  private String myPackageName;
+  private boolean myIsOneTheFly;
+
+  protected GoMultiplePackagesQuickFix(@NotNull GoPackageClause packageClause, Collection<String> packages, boolean isOnTheFly) {
+    super(packageClause);
+    myPackages = packages;
+    myPackageName = ((GoFile)packageClause.getContainingFile()).getPackageName();
```

Comment:
Since packageClause is passed only for retrieving packageName it's better to pass packageName